libs/np-syslib/FormatPrinter.cpp

```diff
@@ -241,7 +245,18 @@ namespace sl
         case FormatSpecifier::String:
             {
                 char* source = va_arg(args, char*);
-                const size_t sourceLength = sl::memfirst(source, 0, 0);
+                size_t sourceLength;
+                if (token.precision == PRECISION_DEFAULT) 
+                {
+                    break;
+                }
+                else if (token.precision != PRECISION_UNSPECIFIED)
+                {
+                    sourceLength = token.precision;
+                } else {
```

Comment:
a hard habit to break, but could you format this like the rest of the code. I.e. else and both of the curly braces on a new line.
If you're worried about space this whole if-else chain could be done without the curly braces, I dont mind :).

Reply:
nono i try to keep using your stile, but sometime i'm just too used the other way! :)